text/0000-rust-for-linux-project-goal-2024h2.md

```diff
@@ -0,0 +1,239 @@
+* [Rust in the linux kernel, by Alice Ryhl](https://www.youtube.com/watch?v=CEznkXjYFb4)
+* [Using Rust in the binder driver, by Alice Ryhl](https://www.youtube.com/watch?v=Kt3hpvMZv8o)
+
+## The next few steps
+
+The RFL project has a [tracking issue][rfl2] listing the unstable features that they rely upon. After discussion with the RFL team, we identified the following subgoals as the ones most urgent to address in 2024. Closing these issues gets us within striking distance of being able to build the RFL codebase on stable Rust.
+
+* Stable support for RFL's customized ARC type
+* Labeled goto in inline assembler and extended `offset_of!` support
+* RFL on Rust CI ([done now!])
+* Pointers to statics in constants [![Owner Needed][]](#ownership-and-other-resources)
+* Custom builds of core/alloc with specialized configuration options [![Owner Needed][]](#ownership-and-other-resources)
+* Code-generation features and compiler options [![Owner Needed][]](#ownership-and-other-resources)
+
+### Stable support for RFL's customized ARC type
+
+One of Rust's great features is that it doesn't "bake in" the set of pointer types.
+The common types users use every day, such as `Box`, `Rc`, and `Arc`, are all (in principle) library defined.
+But in reality those types enjoy access to some unstable features that let them be used more widely and ergonomically.
+Since few users wish to define their own smart pointer types, this is rarely an issue and there has been relative little pressure to stabilize those mechanisms.
+
+The RFL project needs to integrate with the Kernel's existing reference counting types and intrusive linked lists.
+To achieve these goals they've created their own variant of [`Arc`][arclk] (hereafter denoted as `rfl::Arc`),
+but this type cannot be used as idiomatically as the `Arc` type found in `libstd` without two features:
+
+* The ability to be used in methods (e.g., `self: rfl::Arc<Self>`), aka "arbitrary self types", specified in [RFC 3519][].
+* The ability to be coerce to dyn types like `rfl::Arc<dyn Trait>` and then support invoking methods on `Trait` through dynamic dispatch.
+    * This requires the use of two unstable traits, `CoerceUnsized` and `DynDispatch`, neither of which are close to stabilization.
+    * However, [RFC 3621][] provides for a "shortcut" -- a stable interface using `derive` that expands to those traits, leaving room to evolve the underlying details.
+
+Our goal for 2024 is to close those gaps, most likely by implementing and stabilizing [RFC 3519][] and [RFC 3621][].
+
+[rfl2]: https://github.com/Rust-for-Linux/linux/issues/2
+[RFC 3519]: https://github.com/rust-lang/rfcs/pull/3519
+[RFC 3621]: https://github.com/rust-lang/rfcs/pull/3621
+
+### Labeled goto in inline assembler and extended `offset_of!` support
+
+These are two smaller extensions required by the Rust-for-Linux kernel support.
+Both have been implemented but more experience and/or may be needed before stabilization is accepted.
```

Comment:
I think something is missing in this sentence after "and/or". Or maybe the "and/or" shouldn't be there?